### PR TITLE
Private/devidask/dpkg lock check

### PR DESCRIPTION
**Pull request by devidask27**
- Added pre-requisite check to check if there is lock on dpkg and apt.
- Added pre-requisite check to check if system is booted with systemd(PID=1)

### CODE REVIEW

**Review by anupbarve — CHANGES_REQUESTED** [review]
Need to add tests for these checks.

File: pkg/platform/debian/debian.go [hunk 1]
```diff
@@ -314,3 +323,36 @@ func (d *Debian) checkNoexecPermission() (bool, error) {
...
+}
+
+func (d *Debian) checkIfSystemIsBootedWithSystemd() (bool, error) {
+	_, err := d.exec.RunWithStdout("bash", "-c", "pidof systemd | grep '1$'")
+	if err != nil {
+		return false, errors.New("PID of systemd is not 1")
+	} else {
+		return true, nil
+	}
+}
```
Comment (anupbarve) [inline pkg/platform/debian/debian.go:358]: Please remove this check from the current PR. 
We are not yet sure if it works.
Also remove the corresponding test cases. 
We will take it up as part of another PR.

Comment (devidask27) [reply pkg/platform/debian/debian.go:358]: Okay

File: pkg/platform/centos/centos.go [hunk 2]
```diff
@@ -313,3 +316,12 @@ func (c *CentOS) checkNoexecPermission() (bool, error) {
...
 }
+
+func (c *CentOS) checkIfSystemIsBootedWithSystemd() (bool, error) {
+	_, err := c.exec.RunWithStdout("bash", "-c", "pidof systemd | grep '1'")
+	if err != nil {
+		return false, errors.New("PID of systemd is not 1")
+	} else {
+		return true, nil
+	}
+}
```
Comment (anupbarve) [inline pkg/platform/centos/centos.go:327]: Also need to remove this one

Comment (devidask27) [reply pkg/platform/centos/centos.go]: Done

**Review by anupbarve — APPROVED** [review]

**Pull request merged**